compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -116,18 +143,21 @@ void addPackBasedPassPipeline(OpPassManager &pm, TilingConfig &tilingConfig) {
   modulePassManager.addPass(createCanonicalizerPass());
   modulePassManager.addPass(createCSEPass());
 
-  // First level packing
-  AMDAIEPackAndTransposeOptions packOptions0;
-  packOptions0.packLevel = 0;
+  // Fuse both levels of pack ops into for loop
+  AMDAIEFusePackIntoLoopOptions fusePackOptions0;
+  fusePackOptions0.fusePackDepth = 2;
+  fusePackOptions0.useSCFFor = true;
   modulePassManager.addNestedPass<func::FuncOp>(
-      createAMDAIEPackAndTransposePass(packOptions0));
+      createAMDAIEFusePackIntoLoopPass(fusePackOptions0));
+  modulePassManager.addPass(createCanonicalizerPass());
+  modulePassManager.addPass(createCSEPass());
 
-  // Promote inputs and result to shared memory
-  AMDAIEBufferizeToAllocationOptions bufferizeOptions0;
-  bufferizeOptions0.memorySpace = 1;
-  bufferizeOptions0.bufferizeOperand = BufferizeOperand::InputOutput;
+  // Promote the inputs to shared memory
+  AMDAIEBufferizeToAllocationOptions bufferizeOptions2;
+  bufferizeOptions2.memorySpace = 1;
+  bufferizeOptions2.bufferizeOperand = BufferizeOperand::DefInput;
   modulePassManager.addNestedPass<func::FuncOp>(
-      createAMDAIEBufferizeToAllocationPass(bufferizeOptions0));
+      createAMDAIEBufferizeToAllocationPass(bufferizeOptions2));
```

Comment:
It's always good to enclose such options within a scope :-
```
{
    AMDAIEBufferizeToAllocationOptions options;
    . . 
}
```

It'd be nice to adopt this style throughout this pipeline.
It's beneficial because we don't keep those options throughout the end of the function - simply use and discard.

Reply:
Done